### ISSUE TITLE
Upgrade API server to use Tekton version >= 0.19

### ISSUE THREAD

**Issue by fenglixa** [kubeflow/kfp-tekton#402]
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]
Since Tekton version v0.19.0 released, it contains the features (e.g custom task) we may need to use. But seems it was blocked by the issue of:
https://github.com/argoproj/argo/issues/4425

And the PR provided:
https://github.com/argoproj/argo/pull/4810

After that, we can upgrade kfp-tekton API Server to Tekton version 0.19.0.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by Tomcli** [comment]
We are trying to remove all the Argo mock api/tests so it doesn't have to depend on the Argo version. Another task for Tekton 0.19 is to update all the k8s function calls to `go-client 0.18.x`. For example,
https://github.com/tektoncd/pipeline/pull/3319/files#diff-3b3ca39720cb7f55cf1c1529cee802b2f712fcbec78de47484eb3fc5f1d0eed5R34

**Comment by fenglixa** [comment]
@Tomcli Do you know when we could upgrade tekton to latest version?
